doc/REST-interface.md

```diff
@@ -47,18 +47,24 @@ The HTTP request and response are both handled entirely in-memory.
 With the /notxdetails/ option JSON response will only contain the transaction hash instead of the complete transaction details. The option only affects the JSON response.
 
 #### Blockheaders
-`GET /rest/headers/<COUNT>/<BLOCK-HASH>.<bin|hex|json>`
+`GET /rest/headers/<BLOCK-HASH>.<bin|hex|json>?count=<COUNT=5>`
```

Comment:
Why only `count` as a query parameter? How about
```
GET /rest/headers?block_hash=&count=&format=

block_hash defaults to tip
count defaults to 5
format defaults to request accept header, JSON if not specified
```

Reply:
I agree and considered that, but decided to omit this for now to keep this PR's scope limited. Would you be okay with this being incorporated in a follow up PR (which I'm happy to take on right away?)

Reply:
To conclude, I agree with promag that `/headers/` and `/blockfilterheaders/` are collection endpoints since we're not obtaining a resource identified by the specified `block_hash` but rather use it as a pagination parameter. I didn't properly realise this at first. 

Given that 1) this PR has had quite a bit of review time already which I'd rather not overhaul too much and 2) these and the other updates (moving `format` and `notxdetails` to a query parameter) make sense to implement together, I'll implement this in a follow up PR (I've already briefly started but it will take a bit more time as I'm thinking about ways to minimise the maintenance burden of keeping backwards compatibility).

Reply:
FWIW, I agree with @promag that this would be more properly following REST best practices, which seems to be the main point of this PR.

Do I understand correctly that you want to keep maintaining the old structure, this structure and then also future structures like suggested above @stickies-v ? That does seem more laborious than necessary to me. You could merge these first parts of changes as intermediary steps and then add follow-ups but I would not "officially" support it by adding it to docs and release notes yet. You could make the follow-up changes and only afterwards add documentation and release notes after everything is final, so that there are at least only on two versions to maintain and not one per follow-up PR since without docs and release notes it can be reasonably assumed that nobody is building a client on top of the intermediary version. 

Alternatively if you really want to include docs I would suggest in this intermediate step instead of marking the old structure as deprecated, add the new structure marked as experimental and add a comment that it is still object to change.

Reply:
I think your approach makes a lot of sense, fjahr. However, since there has already been significant review time spent on this and this PR seems ready for merge, I don't think this is big enough of a problem to ask for another round of review. Even if it's not a huge update, it still would touch quite a few commits (dropping release notes, modifying REST interface docs, updating error return messages and inline docs). I'll definitely consider the approach you suggest for further PRs because I think it is a more careful approach, but for now I'd prefer to leave things as they are and move forward.